### DISCUSSION TITLE
Is there a way to have default windows already open on startup?

### DISCUSSION THREAD

**Discussion by EZEDSEA** [coder/code-server#2869] [General]
When you start up code-server, is there a way to have a certain files or extensions (i.e. browser preview) already opened in the new session?

**Comment by bpmct** [comment]
This is not something VS Code or code-server runs natively, but I believe that you could use the "Auto Run Command" extension to do this. If you do, would love to hear how it works as I would like to do something similar and then save the settings for other projects

**Reply by jsjoeio** [reply]
Reiterating what Ben said, not something native to code-server but there is an open feature request that looks similar to what you want: https://github.com/cdr/code-server/issues/2594

**Reply by EZEDSEA** [reply]
Thanks for the quick reply guys.  I was wondering this because I've noticed that VS Code (and code-server) already knows how to save your session state when you've opened up various windows, etc.  When you close the app and open it again, your previous files and windows are already opened.

I was hoping this would be a configuration file somewhere.  That one can simply replicate and copy in the respective location.

**Reply by jsjoeio** [reply]
Yeah I know what you mean. And maybe there is one, but we don't know about it. I did a quick search but only found these:
- [VSCode open specific file on startup](https://stackoverflow.com/questions/43851364/vscode-open-specific-file-on-startup)
- [Visual Studio Code always reopens previous file or folder](https://stackoverflow.com/a/36797180/3015595)

Like Ben mentioned, I found [Auto Run Command](https://marketplace.visualstudio.com/items?itemName=gabrielgrinberg.auto-run-command) and [Project Dashboard](https://marketplace.visualstudio.com/items?itemName=kruemelkatze.vscode-dashboard). Not sure if those could help.

**Reply by EZEDSEA** [reply]
Haha. Looks like we're googling the same things... 

I noticed there is state kept in `~/.local/share/code-server/User/state` 

The files are bit hard to read though.... So not the friendliest DX.

**Reply by jsjoeio** [reply]
Yeah those are not super friendly 😅 Sorry we don't have an immediate solution but maybe one of those extensions will unblock you for now. Feel free to chime in on that issue above if you have more to add! Otherwise, you can subscribe to it and get notified when we get to it!

**Reply by bpmct** [reply]
Did you manage to get this working? This is something I've been playing around with (unsuccessfully)
